bin/improver-probabilities-to-realizations

```diff
@@ -45,26 +47,100 @@ def main():
+    # raw ens missing.
+    group = parser.add_mutually_exclusive_group(required=True)
 
-    args = ArgParser(**cli_definition).parse_args()
+    group.add_argument('--reordering', default=False, action='store_true',
+                       help='The option used to create ensemble realizations '
+                       'from percentiles by reordering the input '
+                       'percentiles based on the order of the '
+                       'raw ensemble forecast.')
+    group.add_argument('--rebadging', default=False, action='store_true',
+                       help='The option used to create ensemble realizations '
+                       'from percentiles by rebadging the input '
+                       'percentiles.')
 
+    # If reordering, we need a raw ensemble forecast.
+    reordering = parser.add_argument_group(
+        'Reordering options', 'Options for reordering the input percentiles '
+        'using the raw ensemble forecast as required to create ensemble '
+        'realizations.')
+    reordering.add_argument('--raw_forecast_filepath',
+                            metavar='RAW_FORECAST_FILE',
+                            help='A path to an raw forecast NetCDF file to be '
+                            'processed. This option is compulsory, if the '
+                            'reordering option is selected.')
+    reordering.add_argument(
+        '--random_seed', default=None,
+        help='Option to specify a value for the random seed for testing '
+             'purposes, otherwise, the default random seed behaviour is '
+             'utilised. The random seed is used in the generation of the '
+             'random numbers used for splitting tied values '
+             'within the raw ensemble, so that the values from the input '
+             'percentiles can be ordered to match the raw ensemble.')
+
+    args = parser.parse_args()
+
+    # CLI argument checking:
+    # Can only do one of reordering or rebadging: if options are passed which
+    # correspond to the opposite method, raise an exception.
+    # Note: Shouldn't need to check that both/none are set, since they are
+    # defined as mandatory, but mutually exclusive, options.
+    if args.rebadging:
+        if ((args.raw_forecast_filepath is not None)
+                or (args.random_seed is not None)):
+            parser.wrong_args_error(
+                'raw_forecast_filepath, random_seed', 'rebadging')
+
+    # Process the data
     cube = load_cube(args.input_filepath)
 
-    cube = GeneratePercentilesFromProbabilities().process(
-            cube, no_of_percentiles=args.no_of_realizations)
-    cube = RebadgePercentilesAsRealizations().process(cube)
+    if args.reordering:
+        if args.raw_forecast_filepath is None:
+            message = ("You must supply a raw forecast filepath if using the "
+                       "reording option.")
+            raise ValueError(message)
```

Comment:
Could this be a `wrong_args_error` rather than here?

Reply:
I don't think so. The `wrong_args_error` is designed to say when you pass in something extra that is not compatible. This is saying something should have been passed in but hasn't. I could factor this out as a new `missing_args_error` instead and put it in argparser, but I think it needs to be a different error message.

Reply:
Ah, I hadn't realised there were different "classes" of error here.  I guess I thought it would be tidier to have all the "wrong argument" type clauses at the top of the CLI, but it's not that much of an issue.